### PR TITLE
2023.6.0

### PR DESCRIPTION
**Pull request by SC426**
[Upstream](https://github.com/dask/dask)
[Changelog
](https://docs.dask.org/en/stable/changelog.html)
### Actions
- Updated version number to 2023.6.0
- Updated dependencies
- Added a linter skip for python_build_tool_in_run
- Removed outdated `run_constrained` section

### CODE REVIEW

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]
You will probably want to coordinate with @ryanskeith on https://github.com/AnacondaRecipes/dask-feedstock/pull/30.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -18,19 +18,24 @@ requirements:
     - pip
     - setuptools
     - wheel
+    - versioneer 0.28
   run:
     - python
-    - bokeh >=2.4.2,<3.0
+    - bokeh >=2.4.2
+    - click >=8.0
+    - cloudpickle >=1.5.0
```
[the remark this answers is not in the document]
Comment (SC426) [reply recipe/meta.yaml]: @JeanChristopheMorinPerso I've removed those dependencies. @ryanskeith and I worked on this PR together.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -18,19 +18,17 @@ requirements:
...
-    - bokeh >=2.4.2,<3.0
+    - bokeh >=2.4.2
     - cytoolz >=0.12.0
     - dask-core {{ version }}
     - distributed {{ version }}
     - jinja2 >=2.10.3
     - lz4 >=4.3.2
     - numpy >=1.21
     - pandas >=1.3
-    - pyarrow >=7.0                  # [not s390x]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:31]: Why remove pyarrow?

**Review by marcoesters — COMMENTED** [review]
It might be worth adding `dask.widgets` to the imports.

Other than that, I concur with the question raised by @JeanChristopheMorinPerso about `pyarrow`. It is an [optional dependency](https://github.com/dask/dask/blob/2023.6.0/pyproject.toml#L67), but we have included it so far. Unless there is a compelling reason, it shouldn't be removed.

File: recipe/meta.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcoesters) [reply recipe/meta.yaml]: If you want to make this abundantly clear, you could add a comment.

File: recipe/meta.yaml (hunk 2 above)
Comment (SC426) [reply recipe/meta.yaml:31]: I must have gotten trigger happy with the delete button. I've added it back now.

**Comment by SC426** [comment]
@marcoesters `pyarrow` and `dask.widgets` have been added.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -75,3 +75,5 @@ extra:
     - pitrou
     - tomaugspurger
     - jrbourbeau
+  skip-lints:
+    - python_build_tool_in_run
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:79]: I think that you can remove this now.

**Review by JeanChristopheMorinPerso — APPROVED** [review]
LGMT. The comment I left should be addressed but is not a blocker.

**Pull request merged**